Test RemoteContinuousQueryLeavingRemoteCacheManagerTest leaks thread

### Describe the bug

The test `RemoteContinuousQueryLeavingRemoteCacheManagerTest` is continuously breaking CI with leaking threads. This seems to happen because an operation times out and fails, which causes the test to never close an additional `RemoteCacheManager ` instance, which leaks the threads after the test has finished.

However, this seems to only come from the legacy client. The issue doesn't show on the new client. I'll update both regardless.

### Expected behavior

The extra `RemoteCachemanager` should stop, and no threads should leak.

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

The XML downloaded from CI contains:

```java
org.infinispan.client.hotrod.exceptions.TransportException:: java.net.SocketTimeoutException: PutOperation{(default), key=user1, value=[B0x82011873616D706C655F62616E6B5F61..[79], flags=6, connection=127.0.0.1/127.0.0.1:34197} timed out after 2000 ms
	at org.infinispan.client.hotrod.impl.Util.rewrap(Util.java:87)
	at org.infinispan.client.hotrod.impl.Util.await(Util.java:58)
	at org.infinispan.client.hotrod.impl.RemoteCacheSupport.put(RemoteCacheSupport.java:196)
	at org.infinispan.client.hotrod.impl.RemoteCacheSupport.put(RemoteCacheSupport.java:186)
	at org.infinispan.client.hotrod.event.RemoteContinuousQueryLeavingRemoteCacheManagerTest.testContinuousQueryRemoveRCM(RemoteContinuousQueryLeavingRemoteCacheManagerTest.java:133)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:230)
	at org.infinispan.commons.test.TestNGLongTestsHook.run(TestNGLongTestsHook.java:24)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:242)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:579)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.access$000(SuiteRunner.java:40)
	at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:489)
	at org.testng.internal.thread.ThreadUtil$1.call(ThreadUtil.java:52)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.net.SocketTimeoutException: PutOperation{(default), key=user1, value=[B0x82011873616D706C655F62616E6B5F61..[79], flags=6, connection=127.0.0.1/127.0.0.1:34197} timed out after 2000 ms
	at org.infinispan.client.hotrod.impl.operations.HotRodOperation.run(HotRodOperation.java:183)
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:156)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
```

And the ThreadLeakChecker XML file:

```java
org.infinispan.commons.test.ThreadLeakChecker$LeakException: Leaked thread: RemoteContinuousQueryLeavingRemoteCacheManagerTest-Client-Async-6-13 &lt;&lt; testng-RemoteContinuousQueryLeavingRemoteCacheManagerTest &lt;&lt; org.infinispan.client.hotrod.event.RemoteContinuousQueryLeavingRemoteCacheManagerTest
	at io.netty.channel.epoll.Native.epollWait(Native Method)
	at io.netty.channel.epoll.Native.epollWait(Native.java:220)
	at io.netty.channel.epoll.Native.epollWait(Native.java:213)
	at io.netty.channel.epoll.EpollEventLoop.epollWaitNoTimerChange(EpollEventLoop.java:308)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:365)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.infinispan.commons.test.ThreadLeakChecker$LeakException: testng-RemoteContinuousQueryLeavingRemoteCacheManagerTest &lt;&lt; org.infinispan.client.hotrod.event.RemoteContinuousQueryLeavingRemoteCacheManagerTest
	at org.infinispan.commons.test.ThreadLeakChecker$ThreadInfoLocal.childValue(ThreadLeakChecker.java:120)
	at org.infinispan.commons.test.ThreadLeakChecker$ThreadInfoLocal.childValue(ThreadLeakChecker.java:117)
	at java.base/java.lang.ThreadLocal$ThreadLocalMap.&lt;init&gt;(ThreadLocal.java:469)
	at java.base/java.lang.ThreadLocal.createInheritedMap(ThreadLocal.java:328)
	at java.base/java.lang.Thread.&lt;init&gt;(Thread.java:749)
	at java.base/java.lang.Thread.&lt;init&gt;(Thread.java:1329)
	at org.infinispan.client.hotrod.impl.async.DefaultAsyncExecutorFactory.lambda$getExecutor$0(DefaultAsyncExecutorFactory.java:42)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.&lt;init&gt;(ThreadPoolExecutor.java:637)
	at java.base/java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:928)
	at java.base/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1364)
	at io.netty.util.internal.ThreadExecutorMap$1.execute(ThreadExecutorMap.java:57)
	at io.netty.util.concurrent.SingleThreadEventExecutor.doStartThread(SingleThreadEventExecutor.java:986)
	at io.netty.util.concurrent.SingleThreadEventExecutor.startThread(SingleThreadEventExecutor.java:955)
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:838)
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute0(SingleThreadEventExecutor.java:827)
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:817)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register(AbstractChannel.java:482)
	at io.netty.channel.SingleThreadEventLoop.register(SingleThreadEventLoop.java:89)
	at io.netty.channel.SingleThreadEventLoop.register(SingleThreadEventLoop.java:83)
	at io.netty.channel.MultithreadEventLoopGroup.register(MultithreadEventLoopGroup.java:86)
	at io.netty.bootstrap.AbstractBootstrap.initAndRegister(AbstractBootstrap.java:339)
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:164)
	at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:125)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelInitializer.createChannel(ChannelInitializer.java:88)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelPool.createAndInvoke(ChannelPool.java:221)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelPool.executeDirectlyIfPossible(ChannelPool.java:165)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelPool.acquire(ChannelPool.java:116)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.fetchChannelAndInvoke(Unknown Source)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.pingServersIgnoreException(Unknown Source)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.start(Unknown Source)
	at org.infinispan.client.hotrod.RemoteCacheManager.actualStart(RemoteCacheManager.java:359)
	at org.infinispan.client.hotrod.RemoteCacheManager.start(RemoteCacheManager.java:315)
	at org.infinispan.client.hotrod.RemoteCacheManager.&lt;init&gt;(RemoteCacheManager.java:174)
	at org.infinispan.client.hotrod.test.InternalRemoteCacheManager.&lt;init&gt;(InternalRemoteCacheManager.java:27)
	at org.infinispan.client.hotrod.event.RemoteContinuousQueryLeavingRemoteCacheManagerTest.testContinuousQueryRemoveRCM(RemoteContinuousQueryLeavingRemoteCacheManagerTest.java:121)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:230)
	at org.infinispan.commons.test.TestNGLongTestsHook.run(TestNGLongTestsHook.java:24)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:242)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:579)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.access$000(SuiteRunner.java:40)
	at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:489)
	at org.testng.internal.thread.ThreadUtil$1.call(ThreadUtil.java:52)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	... 3 more
Caused by: org.infinispan.commons.test.ThreadLeakChecker$LeakException: org.infinispan.client.hotrod.event.RemoteContinuousQueryLeavingRemoteCacheManagerTest
```

Observe the operation failed in a line before the extra remote cache manager is stopped.

--- a/client/hotrod-client-legacy/src/test/java/org/infinispan/client/hotrod/event/RemoteContinuousQueryLeavingRemoteCacheManagerTest.java
+++ b/client/hotrod-client-legacy/src/test/java/org/infinispan/client/hotrod/event/RemoteContinuousQueryLeavingRemoteCacheManagerTest.java
@@ -26,6 +26,7 @@ import org.infinispan.client.hotrod.test.MultiHotRodServersTest;
 import org.infinispan.commons.api.query.ContinuousQuery;
 import org.infinispan.commons.api.query.ContinuousQueryListener;
 import org.infinispan.commons.api.query.Query;
+import org.infinispan.commons.time.ControlledTimeService;
 import org.infinispan.commons.time.TimeService;
 import org.infinispan.configuration.cache.CacheMode;
 import org.infinispan.configuration.cache.ConfigurationBuilder;
@@ -33,7 +34,6 @@ import org.infinispan.protostream.SerializationContextInitializer;
 import org.infinispan.query.dsl.embedded.testdomain.User;
 import org.infinispan.query.remote.impl.filter.IckleContinuousQueryProtobufCacheEventFilterConverterFactory;
 import org.infinispan.test.TestingUtil;
-import org.infinispan.commons.time.ControlledTimeService;
 import org.infinispan.util.KeyValuePair;
 import org.testng.annotations.Test;
 
@@ -117,10 +117,6 @@ public class RemoteContinuousQueryLeavingRemoteCacheManagerTest extends MultiHot
    }
 
    public void testContinuousQueryRemoveRCM() {
-      // Create an additional remote cache manager that registers the same query
-      RemoteCacheManager extraRemoteCacheManager = new InternalRemoteCacheManager(createHotRodClientConfigurationBuilder(server(0)).build());
-      RemoteCache<String, User> extraRemoteCache = extraRemoteCacheManager.getCache();
-
       User user1 = new UserPB();
       user1.setId(1);
       user1.setName("John");
@@ -130,22 +126,26 @@ public class RemoteContinuousQueryLeavingRemoteCacheManagerTest extends MultiHot
       user1.setAccountIds(new HashSet<>(Arrays.asList(1, 2)));
       user1.setNotes("Lorem ipsum dolor sit amet");
 
-      remoteCache.put("user" + user1.getId(), user1);
+      final Listener listener;
 
-      Listener listener = applyContinuousQuery(remoteCache);
+      // Create an additional remote cache manager that registers the same query
+      try (RemoteCacheManager extraRemoteCacheManager = new InternalRemoteCacheManager(createHotRodClientConfigurationBuilder(server(0)).build())) {
+         RemoteCache<String, User> extraRemoteCache = extraRemoteCacheManager.getCache();
 
-      // Also register the query on the extra remote cache
-      Listener extraListener = applyContinuousQuery(extraRemoteCache);
+         remoteCache.put("user" + user1.getId(), user1);
 
-      expectElementsInQueue(listener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
-      expectElementsInQueue(extraListener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
-      expectElementsInQueue(listener.updated, 0);
-      expectElementsInQueue(extraListener.updated, 0);
-      expectElementsInQueue(listener.left, 0);
-      expectElementsInQueue(extraListener.left, 0);
+         listener = applyContinuousQuery(remoteCache);
 
-      // Now we shut down the extra remote cache
-      extraRemoteCacheManager.stop();
+         // Also register the query on the extra remote cache
+         Listener extraListener = applyContinuousQuery(extraRemoteCache);
+
+         expectElementsInQueue(listener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
+         expectElementsInQueue(extraListener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
+         expectElementsInQueue(listener.updated, 0);
+         expectElementsInQueue(extraListener.updated, 0);
+         expectElementsInQueue(listener.left, 0);
+         expectElementsInQueue(extraListener.left, 0);
+      }
 
       user1.setAge(23);
       remoteCache.put("user" + user1.getId(), user1);

--- a/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/RemoteContinuousQueryLeavingRemoteCacheManagerTest.java
+++ b/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/RemoteContinuousQueryLeavingRemoteCacheManagerTest.java
@@ -116,10 +116,6 @@ public class RemoteContinuousQueryLeavingRemoteCacheManagerTest extends MultiHot
    }
 
    public void testContinuousQueryRemoveRCM() {
-      // Create an additional remote cache manager that registers the same query
-      RemoteCacheManager extraRemoteCacheManager = new RemoteCacheManager(createHotRodClientConfigurationBuilder(server(0)).build());
-      RemoteCache<String, User> extraRemoteCache = extraRemoteCacheManager.getCache();
-
       User user1 = new UserPB();
       user1.setId(1);
       user1.setName("John");
@@ -129,22 +125,26 @@ public class RemoteContinuousQueryLeavingRemoteCacheManagerTest extends MultiHot
       user1.setAccountIds(new HashSet<>(Arrays.asList(1, 2)));
       user1.setNotes("Lorem ipsum dolor sit amet");
 
-      remoteCache.put("user" + user1.getId(), user1);
+      final Listener listener;
 
-      Listener listener = applyContinuousQuery(remoteCache);
+      // Create an additional remote cache manager that registers the same query
+      try (RemoteCacheManager extraRemoteCacheManager = new RemoteCacheManager(createHotRodClientConfigurationBuilder(server(0)).build())) {
+         RemoteCache<String, User> extraRemoteCache = extraRemoteCacheManager.getCache();
 
-      // Also register the query on the extra remote cache
-      Listener extraListener = applyContinuousQuery(extraRemoteCache);
+         remoteCache.put("user" + user1.getId(), user1);
 
-      expectElementsInQueue(listener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
-      expectElementsInQueue(extraListener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
-      expectElementsInQueue(listener.updated, 0);
-      expectElementsInQueue(extraListener.updated, 0);
-      expectElementsInQueue(listener.left, 0);
-      expectElementsInQueue(extraListener.left, 0);
+         listener = applyContinuousQuery(remoteCache);
 
-      // Now we shut down the extra remote cache
-      extraRemoteCacheManager.stop();
+         // Also register the query on the extra remote cache
+         Listener extraListener = applyContinuousQuery(extraRemoteCache);
+
+         expectElementsInQueue(listener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
+         expectElementsInQueue(extraListener.joined, 1, (kv) -> kv.getValue().getAge(), 22);
+         expectElementsInQueue(listener.updated, 0);
+         expectElementsInQueue(extraListener.updated, 0);
+         expectElementsInQueue(listener.left, 0);
+         expectElementsInQueue(extraListener.left, 0);
+      }
 
       user1.setAge(23);
       remoteCache.put("user" + user1.getId(), user1);

